### PR TITLE
Foia-280: Agency component typeahead search.

### PR DESCRIPTION
**Pull request by jesseday**
* Adds a new typeahead search component and wrapper for the
annual report data form.
* Adds selectedAgencies state to the agency_component data store.
This probably will want to move to our form store once we have that.
* Renders search fields based on data in the selectedAgencies state.
* Renders a "Select Agency Components" button / modal if the selected
agency/component in a search field is an agency with multiple components.
* Builds modal content of checkboxes.
* Adds events to add agencies or components to the selectedAgencies
state and handles them in the agency_component store.
* Adds an AddLink component that adds a blank agency to the store
to render a blank search field.

### CODE REVIEW

File: js/components/foia_report_form_section_one.jsx [hunk 1]
```diff
@@ -66,4 +82,22 @@ class FoiaReportFormSectionOne extends React.Component {
     );
   }
 }
+
+FoiaReportFormSectionOne.propTypes = {
+  /* eslint-disable react/no-unused-prop-types */
```
Comment (bemarlan) [inline js/components/foia_report_form_section_one.jsx]: Can we removed these comments?

File: js/components/foia_report_form_section_one.jsx [hunk 2]
```diff
@@ -66,4 +82,22 @@ class FoiaReportFormSectionOne extends React.Component {
...
+  agencies: PropTypes.instanceOf(Map),
+  agencyComponents: PropTypes.instanceOf(List),
+  selectedAgencies: PropTypes.array,
+  /* eslint-enable react/no-unused-prop-types */
```
Comment (bemarlan) [inline js/components/foia_report_form_section_one.jsx]: Can we removed these comments?

File: js/components/foia_report_form_section_one.jsx (hunk 1 above)
Comment (jesseday) [reply js/components/foia_report_form_section_one.jsx]: Removed from the `FoiaReportFormSectionOne` and `ReportAgencyComponentFilter` components.  I left them in the `ReportAgencyComponentTypeahead` component.  In that component the props are used in componentDidMount, but not in the render method and eslint is complaining about them not being used.

File: js/components/foia_report_form_section_one.jsx (hunk 2 above)
Comment (jesseday) [reply js/components/foia_report_form_section_one.jsx]: Removed from the `FoiaReportFormSectionOne` and `ReportAgencyComponentFilter` components.  I left them in the `ReportAgencyComponentTypeahead` component.  In that component the props are used in componentDidMount, but not in the render method and eslint is complaining about them not being used.  This matches the landing page typeahead component `AgencyComponentFinder`.

**Pull request merged**